### PR TITLE
Rework/modern cleanup

### PR DESCRIPTION
**Pull request by bartdebever**
Rework models, api and server to have modern C# 8-9 features and general improvements.

~~UNTESTED~~

Properly tested now. features tested bellow.

### CODE REVIEW

**Comment by bartdebever** [comment]
Tested:
- Custom effect
- Saving settings
- Schedules
- Overlapping triggers
- Screen mirror
- Device discovery
- Adding a new device
- Layout loading correctly
- Location API

File: Winleafs.Api/Endpoints/ExternalControlEndpoint.cs [hunk 1]
```diff
@@ -33,16 +33,18 @@ public async Task<ExternalControlInfo> GetExternalControlInfoAsync(DeviceType de
...
-                case DeviceType.Aurora:
-                    return await SendRequestAsync<ExternalControlInfo>(BaseUrl, Method.PUT, body: "{\"write\": {\"command\": \"display\", \"animType\": \"extControl\"}}");
-                case DeviceType.Canvas:
-                case DeviceType.Shapes:
-                    return await SendRequestAsync<ExternalControlInfo>(BaseUrl, Method.PUT, body: "{\"write\": {\"command\": \"display\", \"animType\": \"extControl\", \"extControlVersion\": \"v2\"}}");
-                default:
-                    throw new NotImplementedException($"No external control info implemented for device type {deviceType.ToString()}");
-            }            
+                DeviceType.Aurora => await SendRequestAsync<ExternalControlInfo>(BaseUrl, Method.PUT,
+                    body: "{\"write\": {\"command\": \"display\", \"animType\": \"extControl\"}}"),
```
Comment (StijnOostdam) [inline Winleafs.Api/Endpoints/ExternalControlEndpoint.cs:39]: Can this command be an object as well?

File: Winleafs.External/ReleaseClient.cs [hunk 2]
```diff
@@ -7,21 +7,23 @@ namespace Winleafs.External
 {
     public class ReleaseClient : IReleaseClient
     {
-        private const string USERNAME = "StijnOostdam";
-        private const string REPOSITORY_NAME = "Winleafs";
+        private const string Username = "StijnOostdam";
```
Comment (StijnOostdam) [inline Winleafs.External/ReleaseClient.cs:10]: Shouldn't the user name be `winleafs`?

Comment (StijnOostdam) [reply Winleafs.External/ReleaseClient.cs:10]: You can test this by changing the version number in `UserSettings`

Comment (bartdebever) [reply Winleafs.External/ReleaseClient.cs]: Tested it by simply just placing a breakpoint and yes, the API call succeeded just fine.

File: Winleafs.Api/Endpoints/ExternalControlEndpoint.cs (hunk 1 above)
Comment (bartdebever) [reply Winleafs.Api/Endpoints/ExternalControlEndpoint.cs]: Agreed and fixed, want te rework these into records but this will do for now.

**Review by StijnOostdam — APPROVED** [review]

**Pull request merged**